apps/ejabberd/src/mod_http_notification.erl

```diff
@@ -0,0 +1,70 @@
+%%%----------------------------------------------------------------------
+%%% File    : mod_notify.erl
+%%% Author  : Baibossynv Valery <baibossynov.valery@gmail.com>
+%%% Purpose : Message passing via http
+%%% Created : 16 Dec 2015 by Baibossynv Valery <baibossynov.valery@gmail.com>
+%%%----------------------------------------------------------------------
+
+-module(mod_http_notification).
+-author("baibossynov.valery@gmail.com").
+-behaviour(gen_mod).
+
+%% API
+-export([start/2, stop/1, on_user_send_packet/3]).
+
+-include("ejabberd.hrl").
+-include("jlib.hrl").
+
+-define(DEFAULT_HTTP_POOL_SIZE, 5).
+-define(DEFAULT_HTTP_HOST, "http://localhost").
+-define(DEFAULT_PREFIX_PATH, <<"/">>).
+
+start(Host, _Opts) ->
+  HttpHost = gen_mod:get_module_opt(Host, ?MODULE, host, ?DEFAULT_HTTP_HOST),
+  PoolSize = gen_mod:get_module_opt(Host, ?MODULE, pool_size, ?DEFAULT_HTTP_POOL_SIZE),
+  ChildMods = [fusco],
+  ChildMF = {fusco, start_link},
+  ChildArgs = {for_all, [HttpHost, []]},
+  {ok, _} = supervisor:start_child(ejabberd_sup,
+    {{http_notification_sup, Host},
+      {cuesport, start_link,
+        [pool_name(Host), PoolSize, ChildMods, ChildMF, ChildArgs]},
+      transient, 2000, supervisor, [cuesport | ChildMods]}),
+  ejabberd_hooks:add(user_send_packet, Host, ?MODULE, on_user_send_packet, 100),
+  ok.
+
+stop(Host) ->
+  %%TODO: delete children of supervisor
+  ejabberd_hooks:delete(user_send_packet, Host, ?MODULE, on_user_send_packet, 100),
+  ok.
+
+on_user_send_packet(From, To, Packet) ->
+  Type = xml:get_tag_attr_s(list_to_binary("type"), Packet),
+  Body = xml:get_path_s(Packet, [{elem, list_to_binary("body")}, cdata]),
```

Comment:
As we are going towards `exml` application instead of the `xml` module I suggest using `exml_query` [1] interface to get the element.

[1]. https://github.com/esl/exml/blob/master/src/exml_query.erl